examples/knative/powershell/kn-ps-harbor-slack/handler.ps1

```diff
@@ -0,0 +1,112 @@
+Function Process-Init {
+   [CmdletBinding()]
+   param()
+   Write-Host "$(Get-Date) - Processing Init`n"
+
+   Write-Host "$(Get-Date) - Init Processing Completed`n"
+}
+
+Function Process-Shutdown {
+   [CmdletBinding()]
+   param()
+   Write-Host "$(Get-Date) - Processing Shutdown`n"
+
+   Write-Host "$(Get-Date) - Shutdown Processing Completed`n"
+}
+
+Function Process-Handler {
+   [CmdletBinding()]
+   param(
+      [Parameter(Position=0,Mandatory=$true)][CloudNative.CloudEvents.CloudEvent]$CloudEvent
+   )
+
+   # Decode CloudEvent
+   try {
+      $cloudEventData = $cloudEvent | Read-CloudEventJsonData -Depth 10
+   } catch {
+      throw "`nPayload must be JSON encoded"
+   }
+
+   try {
+      $jsonSecrets = ${env:SLACK_SECRET} | ConvertFrom-Json
+   } catch {
+      throw "`nK8s secrets `$env:SLACK_SECRET does not look to be defined"
+   }
+
+   if(${env:FUNCTION_DEBUG} -eq "true") {
+      Write-Host "$(Get-Date) - DEBUG: K8s Secrets:`n${env:SLACK_SECRET}`n"
+
+      Write-Host "$(Get-Date) - DEBUG: CloudEventData`n $(${cloudEventData} | Out-String)`n"
+   }
+
+   # Construct Slack message object
+   $payload = @{
+      attachments = @(
+         @{
+            pretext = $(${jsonSecrets}.SLACK_MESSAGE_PRETEXT);
+            fields = @(
+               @{
+                  title = "Event Type";
+                  value = $cloudEventData.type;
+                  short = "false";
+               }
+               @{
+                     title = "DateTime in UTC";
+                     value = $cloudEvent.time;
+                     short = "false";
+               }
+               @{
+                     title = "Username";
+                     value = $cloudEventData.operator;
+                     short = "false";
+               }
+               @{
+                  title = "Repository Name";
```

Comment:
question: have you tested this with a notification event which might not have these fields set? does the function just produce empty values then or blow up with an exception?

Reply:
@embano1 
Just `empty` values as you can see in this example for `repo_type`, `resource_url` and `operator`:

```json
06/27/2022 09:40:25 - DEBUG: "{
  "attachments": [
    {
      "footer_icon": "https://raw.githubusercontent.com/vmware-samples/vcenter-event-broker-appliance/development/logo/veba_icon_only.png",
      "footer": "Powered by VEBA",
      "pretext": ":harbor: Harbor Slack Function :veba_official:",
      "fields": [
        {
          "short": "false",
          "value": "PUSH_ARTIFACT",
          "title": "Event Type"
        },
        {
          "short": "false",
          "value": "2022-06-25T11:42:42+00:00",
          "title": "DateTime in UTC"
        },
        {
          "short": "false",
          "value": "",
          "title": "Username"
        },
        {
          "short": "false",
          "value": "veba-webhook/bitnami-nginx",
          "title": "Repository Name"
        },
        {
          "short": "false",
          "value": "",
          "title": "Repository Type"
        },
        {
          "short": "false",
          "value": "1.21.6-debian-10-r117",
          "title": "Image Tag"
        },
        {
          "short": "false",
          "value": "",
          "title": "Image Resource Data"
        },
        {
          "short": "false",
          "value": "sha256:d3890814cc5a7cfc02403435281cdf51adfb6b67e223934d9d6137a4ad364286",
          "title": "Image Digest"
        }
      ]
    }
  ]
}"
06/27/2022 09:40:25 - Sending Webhook payload to Slack ...
06/27/2022 09:40:26 - Successfully sent Webhook ...
```

Reply:
ok, perhaps leave a note in the function that for certain events not all fields are present and the slack message could miss details